hazelcast/src/main/java/com/hazelcast/internal/partition/operation/PartitionReplicaSyncRequest.java

```diff
@@ -81,42 +82,60 @@ public void beforeRun() throws Exception {
     }
 
     @Override
-    public void run() throws Exception {
-        NodeEngineImpl nodeEngine = (NodeEngineImpl) getNodeEngine();
-        InternalPartitionServiceImpl partitionService = (InternalPartitionServiceImpl) nodeEngine.getPartitionService();
-        int partitionId = getPartitionId();
-        int replicaIndex = getReplicaIndex();
-
+    public void run() {
+        ILogger logger = getLogger();
+        InternalPartitionServiceImpl partitionService = getService();
         if (!partitionService.areMigrationTasksAllowed()) {
-            ILogger logger = getLogger();
             if (logger.isFinestEnabled()) {
-                logger.finest("Migration is paused! Cannot run replica sync -> " + toString());
+                logger.finest("Migration is paused! Cannot process request. partitionId="
+                        + getPartitionId() + ", replicaIndex=" + getReplicaIndex() + ", namespaces=" + namespaces);
             }
             sendRetryResponse();
             return;
         }
 
-        if (!preCheckReplicaSync(nodeEngine, partitionId, replicaIndex)) {
+        final int permits = acquirePermits();
+        if (permits == 0) {
+            if (logger.isFinestEnabled()) {
+                logger.finest("Not enough permits available! Cannot process request. partitionId="
```

Comment:
Minor: we are going to log this even when the target is wrong.

Reply:
hmm right. I missed that during refactor. When wrong target is detected, another log is printed already. I'll fix that.